src/databricks/labs/ucx/azure/resources.py

```diff
@@ -94,6 +115,18 @@ class AzureRoleAssignment:
     role_name: str
 
 
+@dataclass
+class AccessConnector:
+    id: str
+    name: str
+    location: str
+    type: str = "Microsoft.Databricks/accessConnectors"
+    identity: dict[str, str] = field(default_factory=dict)
```

Comment:
```suggestion
    principal_id: str  # user-assigned and system-assigned
    managed_identity_id: str | None = None # user-assigned
    client_id: str | None = None # user-assigned
    tenant_id: str | None = None # system-assigned
```

doing key-value for `identity` block is not ideal. let's do concrete fields with concrete types and match the `AzureManagedIdentityRequest` structure.

Reply:
user assigned - take the (first) key and add it to `managed_identity_id` property:
<img width="1081" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/5817e15a-69a9-4137-8c50-5a5c343c7692">

system assigned:
<img width="938" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/afaf8140-3039-421e-a6d0-9a038d282f3d">